### PR TITLE
BC-4590 - Refactor filecontentelement

### PR DESCRIPTION
**Pull request by bischofmax**
# Short Description

## Links to Ticket and related Pull-Requests

https://ticketsystem.dbildungscloud.de/browse/BC-4590

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team
- [x] DEV: Every new component is implemented having accessibility in mind (e.g. aria-label, role property)

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

**Comment by bergatco** [comment]
Just checking with the JIRA ticket https://ticketsystem.dbildungscloud.de/browse/BC-4590, I would like to ask, what happened to the mentioned subfolder structure. From my point of view, we agreed on creating a separate subfolder per file content element type (e.g. `image-file-element` or `default-file-element`) and move all specific stuff related to these kind of types into the respective sub folder. Only general components should be placed in the top folder.

**Comment by bergatco** [comment]
I found a lot of occurences of the "old" handling of icons. Is changing the usage / import of icons to the "new" style also part of this ticket?

File: src/components/feature-board-file-element/components/FileDisplay.unit.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+	it("should be found in dom", () => {
+		const { wrapper } = setup();
+
+		const fileDisplay = wrapper.findComponent(FileDisplay);
+
+		expect(fileDisplay.exists()).toBe(true);
+	});
+
+	it("should render image display component with proper props", () => {
```
Comment (bergatco) [inline src/components/feature-board-file-element/components/FileDisplay.unit.ts:38]:
```suggestion
	it("should pass correct props to image display component", () => {
```

File: src/components/feature-board-file-element/components/FileDisplay.unit.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+		const { wrapper, fileNameProp, previewUrlProp, editModeProp } = setup();
+
+		const props = wrapper.findComponent(ImageDisplay).attributes();
+
+		expect(props.filename).toBe(fileNameProp);
+		expect(props.previewUrl).toBe(previewUrlProp);
+		expect(props.editMode).toBe(editModeProp);
+	});
+
+	it("should render default display component with proper props", () => {
```
Comment (bergatco) [inline src/components/feature-board-file-element/components/FileDisplay.unit.ts:48]:
```suggestion
	it("should pass correct props to default display component", () => {
```

File: src/components/feature-board-file-element/components/FileDisplay.unit.ts
Comment (bergatco) [inline src/components/feature-board-file-element/components/FileDisplay.unit.ts:1]: All tests here are failing.

File: src/components/feature-board-file-element/components/FileDisplay.vue [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<template>
+	<div>
+		<ImageDisplay
+			v-if="fileProperties.previewUrl"
```
Comment (bergatco) [inline src/components/feature-board-file-element/components/FileDisplay.vue:4]: Right now it is not needed exactly, but shouldn't we prepare some kind of type differentiation here in order to have it available, when adding different kind of "displays" later on?

File: src/components/feature-board-file-element/components/FilePicker.vue
Comment (bergatco) [inline src/components/feature-board-file-element/components/FilePicker.vue:1]: No tests here as this is basically only a wrapper for a vuetify component and we expect vuetify to do their "homework" (QA) correctly?

File: src/locales/de.json
Comment (bergatco) [inline src/locales/de.json:1]: Why only adapt `de.json`? What about the other languages?

**Comment by bergatco** [comment]
Added some suggestion for a suitable folder structure to https://ticketsystem.dbildungscloud.de/browse/BC-4590 . Please have a look!

File: src/components/feature-board-file-element/components/FileContentElement.unit.ts [hunk 4]
```diff
@@ -142,47 +170,37 @@ describe("FileContentElement", () => {
...
-					await wrapper.vm.$nextTick();
+				expect(card.props("outlined")).toBe(false);
+			});
 
-					const fileContentElementDisplay = wrapper.findComponent(
-						FileContentElementDisplay
-					);
-					expect(fileContentElementDisplay.exists()).toBe(true);
-				});
+			describe("when v-card emits keydown.up.down event", () => {
```
Comment (bischofmax) [inline src/components/feature-board-file-element/components/FileContentElement.unit.ts:185]: @bergatco This tests currently fails. I saw your comment on this in the old tests. But I didn't get why we could not make it working. Could you have a look, please?

Comment (bergatco) [reply src/components/feature-board-file-element/components/FileContentElement.unit.ts]: Fixed! 😉

File: src/components/feature-board-file-element/content/display/image-display/ImageDisplay.vue [hunk 5]
```diff
@@ -0,0 +1,30 @@
+<template>
+	<div>
```
Comment (SevenWaysDP) [inline src/components/feature-board-file-element/content/display/image-display/ImageDisplay.vue:2]: remove please

File: src/components/feature-board-file-element/content/FileContent.unit.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
+import { PreviewStatus } from "@/fileStorageApi/v3";
+import createComponentMocks from "@@/tests/test-utils/componentMocks";
+import { shallowMount } from "@vue/test-utils";
+import FileAlert from "./alert/FileAlert.vue";
+import FileContent from "./FileContent.vue";
+import ContentElementFooter from "./footer/ContentElementFooter.vue";
+
+describe("FileContent", () => {
+	const setup = () => {
```
[the remark this answers is not in the document]
Comment (bischofmax) [reply src/components/feature-board-file-element/content/FileContent.unit.ts:9]: when? :-) There is no logic in the component with different behavior on varying scenarios.

**Review by bergatco — APPROVED** [review]

**Review by SevenWaysDP — APPROVED** [review]

**Pull request merged**